### ISSUE TITLE
Multiple @javascript tests break zombie.js

### ISSUE THREAD

**Issue by IslandUsurper** [linusnorton/grunt-parallel-behat#12]
I've been writing tests in Behat for my Drupal site, and I've based my setup after the example/ folder from grunt-drupal-tasks, so I'm running tests with Goutte and Zombie.js. However, as soon as I had a second Feature tagged `@javascript`, it throws this error:

```
      Server process has been terminated: (8) [
      events.js:72
              throw er; // Unhandled 'error' event
                    ^
      Error: listen EADDRINUSE
          at errnoException (net.js:904:11)
          at Server._listen2 (net.js:1042:14)
          at listen (net.js:1064:10)
          at net.js:1146:9
          at Object._onImmediate (node_modules/zombie/lib/zombie/dns_mask.js:69:16)
          at processImmediate [as _immediateCallback] (timers.js:345:15)
      ] (RuntimeException)
```

I thought maybe it was an issue in Mink, and it kind of is, because of the way they call zombie.js, but they said Mink accepts setting the port in the environment (see minkphp/MinkZombieDriver#126). So, the thing to do appears to be using a different port for each test process that gets spawned. Considering how prevalent JavaScript is, I think it would be prudent to make this the default behavior.

**Comment by linusnorton** [comment]
I think rather than have grunt-parallel-behat select a random port which may or may not be in use by another process it might be better to add a configuration option to allow a list of ports to use. In theory you would be able to specify:

```
ports: [4004, 5000, 5020, 4050]
```

and it could cycle through them. We would need to track which ports are currently in use by a behat instance. Another issue to think about would be that if the number if tests run in parallel exceeds the number of ports available.

**Comment by IslandUsurper** [comment]
It should be really easy to check that `maxProcesses <= ports.length`. Or you could actually set maxProcesses to that amount.

I don't think that's the biggest problem. grunt-parallel-behat doesn't know or care which driver is used in the tests. But it has to pass the port into a specific driver's configuration at the time it runs the test, and it goes in different places for Selenium and Zombie, and not even used by Goutte. Figuring out when and how to inject the port specification requires opening up the behat.yml file to determine what "javascript_session" is set to, or would be if it isn't set. Maybe we can get away with not inspecting the feature files for `@javascript` tags, since all of the drivers will be activated anyway.

**Comment by berliner** [comment]
Any progress on this?

**Comment by linusnorton** [comment]
Unfortunately not, I don't have a test suite available that I can test this with. I'd gladly accept a pull request though.

**Comment by berliner** [comment]
I'm very bad with javascript, but I came up with a first draft that works for me in my context, see 1040c6ea1d8d5af06f7df7ee86a2648cf5c612b5

I'm using mink and zombiejs, so that's all that is supported at the moment

**Comment by linusnorton** [comment]
@berliner the patch looks fine to me, can you make a PR and check the tests are passing?

**Comment by berliner** [comment]
Tests, good point, I'll check these first before issuing a PR.

**Comment by linusnorton** [comment]
@berliner shall we close this issue now?

**Comment by berliner** [comment]
@linusnorton Yes, it's fine with me. Parallel execution works now in my use case. Thanks for the quick and the follow up!
